### ISSUE TITLE
Misleading error message with empty trait

### ISSUE THREAD

**Issue by frewsxcv** [rust-lang/rust#23689]
`/tmp $ cat program.rs`

```rust
trait MyTrait {}

fn main() {}
```

`/tmp $ rustc program.rs`

```
program.rs:1:1: 1:17 error: parameter `Self` is never used
program.rs:1 trait MyTrait {}
             ^~~~~~~~~~~~~~~~
program.rs:1:17: 1:17 help: consider removing `Self` or using a marker such as `core::marker::PhantomFn`
error: aborting due to previous error
```

My program does not make use of any `Self` parameter

**Comment by rprichard** [comment]
Looks like a duplicate of #22598.
